server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -323,16 +323,11 @@ public static ThreadPoolExecutor createSelfResizingThreadPool(final String serve
       // however, this isn't really an issue, since it adjusts periodically anyway
       if (pool.getCorePoolSize() <= pool.getActiveCount()) {
         int larger = pool.getCorePoolSize() + Math.min(pool.getQueue().size(), 2);
-        log.info("Increasing server thread pool size on {} to {}", serverName, larger);
-        pool.setMaximumPoolSize(larger);
-        pool.setCorePoolSize(larger);
+        ThreadPools.resizePool(pool, () -> larger, serverName + "-ClientPool");
       } else {
         if (pool.getCorePoolSize() > pool.getActiveCount() + 3) {
           int smaller = Math.max(executorThreads, pool.getCorePoolSize() - 1);
-          if (smaller != pool.getCorePoolSize()) {
-            log.info("Decreasing server thread pool size on {} to {}", serverName, smaller);
-            pool.setCorePoolSize(smaller);
-          }
+          ThreadPools.resizePool(pool, () -> smaller, serverName + "-ClientPool");
```

Comment:
So, it would [appear](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/util/concurrent/ThreadPoolExecutor.java#L1533) that lowering the corePoolSize below the MaxPoolSize throws an error. Maybe this is what you were referring to?

Reply:
I just noticed that this code was no longer making the same method calls on the thread pool that it used to make, however I was not sure if that change mattered.

Reply:
Do you think the existing code before your change had a bug?

Reply:
Right, it's now using the common way of doing the resizing.

Reply:
hmm so maybe this code never achieved its intent.. I wonder if it only increased but never decreased

Reply:
I can't say about before my change, but in ThreadPools.java as it is now, if the thread pool was created with a pool size of 5 and no max pool specified, then the max pool size would also be set to 5. If the core pool size was then decreased, it would error.